### PR TITLE
fix: detect s390x arch and exclude noasm tag

### PR DESCRIPTION
**Pull request by jonathan-albrecht-ibm**
Commit https://github.com/influxdata/influxdb/commit/074cd24a60ddd5aa2a851a7d04faaef5c126562a added a fix to compile with the go noasm build tag for all non-amd64 arch because arm64 tests panic without it. This also makes s390x compile with noasm even though the tests run fine on s390x without noasm. 
This change adds s390x to the list of architectures that should not set noasm.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by danxmoran** [comment]
Hi @jonathan-albrecht-ibm, thanks for the submission! I see in https://github.com/apache/arrow/commit/87194de72e9b09ea72ca96a218f3dc67ed5f9547 that you recently fixed a problem building s390x in Arrow. Do we need to wait until flux bumps its Arrow dependency before merging this change?

**Comment by jonathan-albrecht-ibm** [comment]
> Hi @jonathan-albrecht-ibm, thanks for the submission! I see in [apache/arrow@87194de](https://github.com/apache/arrow/commit/87194de72e9b09ea72ca96a218f3dc67ed5f9547) that you recently fixed a problem building s390x in Arrow. Do we need to wait until flux bumps its Arrow dependency before merging this change?

No, I actually ran into the Arrow problem (arrow not able to build with noasm) because Influx _is_ building with noasm on s390x. So the arrow problem won't be hit after this change and no need to wait.

**Comment by jonathan-albrecht-ibm** [comment]
gentle nudge: Just wondering about the status of this pr and if there is anything I can do?

**Comment by danxmoran** [comment]
Apologies for the delay @jonathan-albrecht-ibm, I was working through some CI issues to make it easier to test changes from forked repos. We should be set to proceed now.

I don't think we intend to "officially" support/release for s390x, but if this patch fixes a problem for you I'm ok merging it :+1: could you please rebase onto the latest `master` branch? There might also be an "allow edits from maintainers" button you could click that would let me update the branch via the UI as needed, so you don't have to keep pushing an update when others merge before you.

**Review by danxmoran — APPROVED** [review]

**Comment by jonathan-albrecht-ibm** [comment]
Thanks for looking at it @danxmoran, I've rebased the PR.

**Comment by jonathan-albrecht-ibm** [comment]
@danxmoran Thanks for reviewing and merging this PR

**Pull request merged**
